datafusion/core/src/physical_optimizer/optimizer.rs

```diff
@@ -97,6 +97,8 @@ impl PhysicalOptimizer {
             // Note that one should always run this rule after running the EnforceDistribution rule
             // as the latter may break local sorting requirements.
             Arc::new(EnforceSorting::new()),
+            // TODO: `try_embed_to_hash_join` in the ProjectionPushdown rule would be block by the CoalesceBatches, so add it before CoalesceBatches. Maybe optimize it in the future.
+            Arc::new(ProjectionPushdown::new()),
```

Comment:
Do we need to have two `ProjectionPushdown`? The original can be removed?

Reply:
AFAICT the original will be modified to account for the new built-in projection capability and this one will be removed

Reply:
> the original will be modified to account for th

Is this refers to #9111?

Reply:
I guess so.